tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -4073,8 +4073,13 @@ void IrEmitterUnnested::EmitReductionOutputForRowReduction(
       llvm::Type* element_type =
           state.partial_result_address->getType()->getElementType();
 
-      llvm::Value* initial_value_addr = llvm_ir::EmitAllocaAtFunctionEntry(
-          element_type, "initial_value_addr", &b_);
+/* Insure initial value address is in generic, not scratch. */ 
```

Comment:
What about all other allocas we use in reduction? Why only this one needs to be wrapped? Or we do need to wrap all of them?

Reply:
The other allocas aren't generating a conflict in the address space in the llvm ir pre-optimization. This EmitAllocaAtFunctionEntry is sending the initial_value_addr to scratch in the AMD ir-gen. In https://github.com/tensorflow/tensorflow/commit/fc9df104fdbbda6ee9096b40186a50badb147ac7#diff-3d7be85f9e38ee47e75201cc69e4d4d5f14a49db5970de4c1205229fbd23d2f0L4121-L4122
there is the cast from Shared to Global.

Reply:
Ah thanks I see! Could you then reinstate `CastSharedToGlobal`? Sorry for the trouble!

Reply:
@cheshire I implemented the requested changes.